generators/entity-server/templates/src/main/java/package/web/rest/EntityResource.java.ejs

```diff
@@ -245,17 +239,19 @@ public class <%= entityClass %>Resource {
   <%_ } _%>
 
   <%_ if (saveUserSnapshot) { _%>
-    <%_ if (manyToManyWithUser) { _%>
-        if (<%= instanceName %>.getUsers() != null) {
-            // Save users in case they are new and only exists in gateway
-            userRepository.saveAll(<%= instanceName %>.getUsers());
-        }
-    <%_ } else { _%>
-        if (<%= instanceName %>.getUser() != null) {
-            // Save user in case it's new and only exists in gateway
-            userRepository.save(<%= instanceName %>.getUser());
-        }
-    <%_ } _%>
+    <% for (const userRelationship of relationships.filter(rel => rel.otherEntity.builtInUser)) { %>
+      <%_ if (userRelationship.collection) { _%> 
+          if (<%= instanceName %>.get<%= userRelationship.relationshipNameCapitalizedPlural %>() != null) { 
+              // Save users in case they are new and only exist in gateway 
+              userRepository.saveAll(<%= instanceName %>.get<%= userRelationship.relationshipNameCapitalizedPlural %>()); 
+          } 
+      <%_ } else { _%> 
+          if (<%= instanceName %>.get<%= userRelationship.relationshipNameCapitalized %>() != null) { 
+              // Save user in case it's new and only exists in gateway 
+              userRepository.save(<%= instanceName %>.get<%= userRelationship.relationshipNameCapitalized %>()); 
+          } 
```

Comment:
```suggestion
        if (<%= instanceName %>.get<%= userRelationship.relationshipNameCapitalizedPlural %>() != null) { 
            // Save users in case they are new and only exist in gateway 
            userRepository.saveAll(<%= instanceName %>.get<%= userRelationship.relationshipNameCapitalizedPlural %>()); 
        } 
      <%_ } else { _%> 
        if (<%= instanceName %>.get<%= userRelationship.relationshipNameCapitalized %>() != null) { 
            // Save user in case it's new and only exists in gateway 
            userRepository.save(<%= instanceName %>.get<%= userRelationship.relationshipNameCapitalized %>()); 
        } 
```